src/service/engine/model.ts

```diff
@@ -1,95 +1,78 @@
 
-  constructor(context: Context, engine: Engine) {
-    const { id, name, description, endpoint, current_status_summary } = engine;
-    this.id = id;
+  constructor(context: Context, connection: Connection, engine: Engine) {
+    const { name, endpoint, current_status_summary } = engine;
     this.name = name;
-    this.description = description;
     this.endpoint = endpoint;
     this.context = context;
+    this.connection = connection;
     this.current_status_summary = current_status_summary;
   }
 
   async start() {
-    const { apiEndpoint, httpClient } = this.context;
-    const id = this.id.engine_id;
-    const accountId = this.context.resourceManager.account.id;
-    const url = `${apiEndpoint}/${ACCOUNT_ENGINE_START(accountId, id)}`;
-    const data = await httpClient
-      .request<{ engine: Engine }>("POST", url)
-      .ready();
-    return data;
+    const query = `START ENGINE ${this.name}`;
+    await this.connection.execute(query);
+    await this.refreshStatus();
+    const res: Engine = {
+      name: this.name,
+      endpoint: this.endpoint,
+      current_status_summary: this.current_status_summary
+    };
+    return { engine: res };
   }
 
   async startAndWait() {
-    const {
-      engine: { current_status_summary }
-    } = await this.start();
-    this.current_status_summary = current_status_summary;
-    if (this.current_status_summary.includes("RUNNING")) {
-      return;
-    }
-
-    let interval: NodeJS.Timer;
-    await new Promise<void>(resolve => {
-      interval = setInterval(async () => {
-        await this.refreshStatus();
-        if (this.current_status_summary.includes("RUNNING")) {
-          return resolve();
-        }
-      }, 10 * 1000); // Check every 10 seconds.
-    }).finally(() => {
-      if (interval) {
-        clearInterval(interval);
-      }
-    });
+    // START ENGINE is now blocking
+    return this.start();
   }
 
   async stop() {
-    const { apiEndpoint, httpClient } = this.context;
-    const id = this.id.engine_id;
-    const accountId = this.context.resourceManager.account.id;
-    const url = `${apiEndpoint}/${ACCOUNT_ENGINE_STOP(accountId, id)}`;
-    const data = await httpClient
-      .request<{ engine: Engine }>("POST", url)
-      .ready();
-    return data;
+    const query = `STOP ENGINE ${this.name}`;
+    await this.connection.execute(query);
+    await this.refreshStatus();
+    const res: Engine = {
+      name: this.name,
+      endpoint: this.endpoint,
+      current_status_summary: this.current_status_summary
+    };
+    return { engine: res };
   }
 
   async restart() {
-    const { apiEndpoint, httpClient } = this.context;
-    const id = this.id.engine_id;
-    const accountId = this.context.resourceManager.account.id;
-    const url = `${apiEndpoint}/${ACCOUNT_ENGINE_RESTART(accountId, id)}`;
-    const data = await httpClient
-      .request<{ engine: Engine }>("POST", url)
-      .ready();
-    return data;
+    // TODO: is this right?
```

Comment:
As we discussed on the meeting, it is right and it should be deprecated from now